### PR TITLE
fix(admin): superuser user put

### PR DESCRIPTION
**Pull request by cathteng**
Correctly parse snakeCase into camel_case

### CODE REVIEW

File: src/sentry/api/endpoints/user_details.py [hunk 1]
```diff
@@ -168,6 +165,18 @@ def put(self, request: Request, user) -> Response:
...
+                return Response(
+                    {"detail": "Missing required permission to add superuser."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+            elif not user.is_staff and request.data.get("isStaff"):
+                return Response(
+                    {"detail": "Missing required permission to add admin."},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
```
Comment (leeandher) [inline src/sentry/api/endpoints/user_details.py:179]: nit: Permission checks usually go before any serialization. Either that, or you can add `validate_is_staff()` and `validate_is_superuser()` methods to the `BaseUserSerializer` and just rely on that for throwing errors.

**Review by leeandher — APPROVED** [review]
looks good! great tests and refactor, small comment to consider, but ready to 🚢

File: src/sentry/api/endpoints/user_details.py [hunk 2]
```diff
@@ -168,6 +165,18 @@ def put(self, request: Request, user) -> Response:
         if not serializer.is_valid() or not serializer_options.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        if not request.access.has_permission("users.admin"):
+            if not user.is_superuser and request.data.get("isSuperuser"):
```
[the remark this answers is not in the document]
Comment (cathyteng17) [reply src/sentry/api/endpoints/user_details.py:169]: This is for explicitly throwing an error. We weren't doing this before and simply returning 200 OK  regardless of whether isSuperuser or isStaff were actually updated.

[the remark this answers is not in the document]
Comment (cathyteng17) [reply src/sentry/api/endpoints/user_details.py:169]: Also, the user being referenced here is the user we're trying to update

**Pull request merged**